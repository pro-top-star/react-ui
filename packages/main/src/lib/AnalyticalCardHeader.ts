import { AnalyticalCardHeader } from '../components/AnalyticalCard/header/AnalyticalCardHeader';

export { AnalyticalCardHeader };
