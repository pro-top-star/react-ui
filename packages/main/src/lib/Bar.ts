import { Bar } from '../components/Bar';

export { Bar };
