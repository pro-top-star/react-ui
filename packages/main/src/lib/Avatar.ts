import { Avatar } from '../webComponents/Avatar';
import type { AvatarPropTypes } from '../webComponents/Avatar';

export { Avatar };
export type { AvatarPropTypes };
