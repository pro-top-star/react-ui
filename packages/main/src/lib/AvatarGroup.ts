import { AvatarGroup } from '../webComponents/AvatarGroup';
import type { AvatarGroupPropTypes } from '../webComponents/AvatarGroup';

export { AvatarGroup };
export type { AvatarGroupPropTypes };
