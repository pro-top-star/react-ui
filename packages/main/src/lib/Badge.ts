import { Badge } from '../webComponents/Badge';
import type { BadgePropTypes } from '../webComponents/Badge';

export { Badge };
export type { BadgePropTypes };
