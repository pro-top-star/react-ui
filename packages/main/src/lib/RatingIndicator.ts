import { RatingIndicator } from '../webComponents/RatingIndicator';
import type { RatingIndicatorPropTypes } from '../webComponents/RatingIndicator';

export { RatingIndicator };
export type { RatingIndicatorPropTypes };
