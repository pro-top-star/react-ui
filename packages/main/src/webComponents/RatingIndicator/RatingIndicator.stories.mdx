import { Title, Subtitle, Description, Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import { RatingIndicator } from '@ui5/webcomponents-react/lib/RatingIndicator';
import { createSelectArgTypes } from '@shared/stories/createSelectArgTypes';

<Meta
  title="UI5 Web Components / RatingIndicator"
  component={RatingIndicator}
  args={{
    value: 3
  }}
/>

<Title />
<Subtitle />
<Description />

<Preview>
  <Story name="Default">
    {(args) => {
      return <RatingIndicator {...args} />;
    }}
  </Story>
</Preview>

<Props story="Default" components={{ RatingIndicator }} />
