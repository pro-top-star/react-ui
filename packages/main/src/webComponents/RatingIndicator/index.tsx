import { withWebComponent, WithWebComponentPropTypes } from '@ui5/webcomponents-react/lib/withWebComponent';
import '@ui5/webcomponents/dist/RatingIndicator';
import { FC } from 'react';

export interface RatingIndicatorPropTypes extends Omit<WithWebComponentPropTypes, 'onChange'> {
  /**
   * Defines whether the <code>RatingIndicator</code> is disabled.<br/><br/>
   *
   * <b>Note:</b> A disabled <code>RatingIndicator</code> is completely noninteractive.
   */
  disabled?: boolean;
  /**
   * The number of displayed rating symbols
   */
  maxValue?: number;
  /**
   * Defines whether the <code>RatingIndicator</code> is read-only. <br><br> <b>Note:</b> A read-only <code>RatingIndicator</code> is not editable, but still provides visual feedback upon user interaction.
   */
  readonly?: boolean;
  /**
   * The indicated value of the rating <br><br> <b>Note:</b> If you set a number which is not round, it would be shown as follows: <ul> <li>1.0 - 1.2 -> 1</li> <li>1.3 - 1.7 -> 1.5</li> <li>1.8 - 1.9 -> 2</li> <ul>
   */
  value?: number;
  /**
   * The event is fired when the value changes.
   */
  onChange?: (event: CustomEvent<{}>) => void;
}

/**
 * The rating indicator is used to display a specific number of icons that are used to rate an item. Additionally, it is
 also used to display the average and overall ratings.

 * <a href="https://sap.github.io/ui5-webcomponents/playground/components/RatingIndicator" target="_blank">UI5 Web Components Playground</a>
 */
const RatingIndicator: FC<RatingIndicatorPropTypes> = withWebComponent<RatingIndicatorPropTypes>(
  'ui5-rating-indicator',
  ['maxValue', 'value'],
  ['disabled', 'readonly'],
  [],
  ['change']
);

RatingIndicator.displayName = 'RatingIndicator';

RatingIndicator.defaultProps = {
  disabled: false,
  maxValue: 5,
  readonly: false,
  value: 0
};

export { RatingIndicator };
