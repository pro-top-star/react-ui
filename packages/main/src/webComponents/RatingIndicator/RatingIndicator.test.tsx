import { RatingIndicator } from '@ui5/webcomponents-react/lib/RatingIndicator';
import { mount } from 'enzyme';
import React from 'react';

describe('RatingIndicator', () => {
  test('Basic Test (generated)', () => {
    const wrapper = mount(<RatingIndicator />);
    expect(wrapper.render()).toMatchSnapshot();
  });
});
