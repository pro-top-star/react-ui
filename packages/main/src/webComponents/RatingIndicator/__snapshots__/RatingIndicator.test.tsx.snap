// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingIndicator Basic Test (generated) 1`] = `
<ui5-rating-indicator
  max-value="5"
  value="0"
/>
`;
